src/vcf/store_parse_policy.cpp

```diff
@@ -161,37 +158,37 @@ namespace ebi
 
     void StoreParsePolicy::handle_body_line(ParsingState const & state) 
     {
+        // Transform the position token into a size_t
+        auto position = static_cast<size_t>(std::stoi(m_line_tokens["POS"][0]));
+
+        // Transform all the quality tokens into floating point numbers
+        float quality;
         try {
-            // Transform the position token into a size_t
-            auto position = static_cast<size_t>(std::stoi(m_line_tokens["POS"][0]));
```

Comment:
`std::stoi` could also raise an `std::invalid_argument` exception, although it should probably be in a different try-catch than the quality check.